arangod/Cluster/UpdateCollection.cpp

```diff
@@ -76,21 +76,36 @@ UpdateCollection::UpdateCollection(MaintenanceFeature& feature, ActionDescriptio
   }
   TRI_ASSERT(desc.has(FOLLOWERS_TO_DROP));
 
+  TRI_ASSERT(desc.has(OLD_CURRENT_COUNTER));
+
   if (!error.str().empty()) {
-    LOG_TOPIC("a6e4c", ERR, Logger::MAINTENANCE) << "UpdateCollection: " << error.str();
+    LOG_TOPIC("a6e4c", ERR, Logger::MAINTENANCE)
+        << "UpdateCollection: " << error.str();
     _result.reset(TRI_ERROR_INTERNAL, error.str());
     setState(FAILED);
   }
 }
 
 void handleLeadership(LogicalCollection& collection, std::string const& localLeader,
-                      std::string const& plannedLeader, std::string const& followersToDrop) {
+                      std::string const& plannedLeader,
+                      std::string const& followersToDrop, std::string const& databaseName,
+                      uint64_t oldCounter, MaintenanceFeature& feature) {
   auto& followers = collection.followers();
 
   if (plannedLeader.empty()) {   // Planned to lead
     if (!localLeader.empty()) {  // We were not leader, assume leadership
       followers->setTheLeader(std::string());
       followers->clear();
+      // This will block the thread until we fetched a new current version
+      // in maintenance main thread.
+      feature.waitForLargerCurrentCounter(oldCounter);
+      auto currentInfo = ClusterInfo::instance()->getCollectionCurrent(
+          databaseName, std::to_string(collection.planId()));
+      // TODO Is this guaranteed? Or can we have a stale current version where
```

Comment:
?

Reply:
I guess I will add a nullptr check here.
Situation is as follows:
DBServer 1 (leader) dies.
DBServer 2 takes over, starts Phase1 => Creates this leadership job, now waits until at least once current is fetched.
Meanwhile: Coordinator orders drop() on the collection, which (for some reason) is faster in dropping the collection, then Phase2 pulling Current.
Now current does not have any information about this collection (it was dropped)
=> I get nullptr here.

Reply:
Forget it, drop is done via a Maintenance cycle, so guarantee is that the collection is either dropped of switching leader, not both in one P1 P2 cycle.